openupgrade_scripts/scripts/sale/16.0.1.2/upgrade_analysis_work.txt

```diff
@@ -0,0 +1,93 @@
+# NOTHING TO DO
+
+sale         / sale.order               / show_update_fpos (boolean)    : NEW
+sale         / sale.order               / show_update_pricelist (boolean): not stored anymore
+# NOTHING TO DO: handle by ORM
+
+sale         / sale.order.line          / analytic_distribution_stored_char (char): NEW isfunction: function, stored
+# NOTHINGH TO DO: handle by ORM
+
+sale         / sale.order.line          / analytic_tag_ids (many2many)  : DEL relation: account.analytic.tag
+# NOTHINGH TO DO: not used any more
+
+sale         / sale.order.line          / product_type (selection)      : previously in module sale_stock
+# DONE: pre-migration: moved from sale_stock to sale
+
+sale         / sale.order.line          / qty_delivered (float)         : not a function anymore
+# NOTHING TO DO: No database change
+
+sale         / sale.order.line          / qty_delivered_manual (float)  : DEL
+# NOTHING TO DO: not used any more, this field is useless since the new compute method from 13.0
+
+---XML records in moduldatabasee 'sale'---
+NEW ir.actions.act_window: sale.action_open_sale_payment_provider_onboarding_wizard
+DEL ir.actions.act_window: sale.action_open_sale_onboarding_payment_acquirer_wizard
+DEL ir.actions.act_window: sale.action_sale_order_form_view
+# NOTHING TO DO: noupdate="0" records
+
+ir.actions.server: sale.model_sale_order_action_share (noupdate switched)
+# DONE: pre-migration: noupdate switched to 1
+
+NEW ir.config_parameter: sale.default_invoice_email_template (noupdate)
+# DONE: pre-migration: add xmlid
+
+NEW ir.model.access: sale.access_product_product_attribute_custom_value_sale_user
+NEW ir.model.access: sale.access_product_tag_sale_manager
+NEW ir.model.access: sale.access_sale_payment_provider_onboarding_wizard
+DEL ir.model.access: sale.access_account_account_sale_manager
+DEL ir.model.access: sale.access_account_account_type_sale_salesman
+DEL ir.model.access: sale.access_account_analytic_tag_sale_salesman
+DEL ir.model.access: sale.access_account_move_manager
+DEL ir.model.access: sale.access_account_tax_sale_manager
+DEL ir.model.access: sale.access_product_product_attribute_custom_value_sale_manager
+DEL ir.model.access: sale.access_report_all_channels_sales
+DEL ir.model.access: sale.access_sale_payment_acquirer_onboarding_wizard
+DEL ir.model.access: sale.access_sale_report_manager
+# NOTHING TO DO
+
+ir.model.constraint: sale.constraint_sale_order_date_order_conditional_required (changed definition: is now 'check((state in('sale','done') and date_order is not null) or state not in('sale','done'))' ('check((state in('sale','done') and date_order is not null) or state not in('sale','done') )'))
+# DONE: pre-migration: dropped constraint and let ORM add it again with the new definition
+
+NEW ir.rule: sale.sale_payment_provider_onboarding_wizard_rule (noupdate)
+DEL ir.rule: sale.sale_payment_acquirer_onboarding_wizard_rule (noupdate)
```

Comment:
the model, the name and the xml id has changed. it is better to simply delete `sale.sale_payment_acquirer_onboarding_wizard_rule`. the new rule will be created automatically anyway.